src/uast_private.h

```diff
@@ -0,0 +1,23 @@
+#ifndef LUAST_SRC_UAST_PRIVATE_H_
+#define LUAST_SRC_UAST_PRIVATE_H_
+
+#include <libxml/tree.h>
+#include "uast.h"
+#include "find-ctx.h"
+// These functions are used internally for testing and not exported.
+
+// Sets the length of a find_ctx and allocates memory for the results array
+// if needed.
+// Returns 0 if the length was changed correctly.
+int find_ctx_set_len(find_ctx *ctx, int len);
+
+// Returns the total capacity of the internal buffer.
+int find_ctx_get_cap(const find_ctx *ctx);
+
+// Returns the pointer to the array of results.
+void **find_ctx_get_results(const find_ctx *ctx);
+
+static xmlDocPtr create_document(node_api *api, void *node);
+static xmlNodePtr create_xml_node(node_api *api, void *node, xmlNodePtr parent);
```

Comment:
Why to add the static methods to a header that is used across different compilation utils (uast.o and tests/main.o)?

Even so the CI passes, it gives the fake sensation that `create_xml_node()` and `create_document()` can be used from any .c that includes this private header (like tests/main) but it can't.

If node_tests tries to use any of those, it will fail with a `"undefined reference to x"`.

I am not a C expert, but I think it is not a good practice to use `static` in headers that can be used across more than one compilation unit.

UPDATED: 
verified it myself: using create_document() in tests/main causes:
```
In file included from /home/manu/repos/sourced/libuast/tests/nodes-test.h:13:0,
                 from /home/manu/repos/sourced/libuast/tests/main.cpp:5:
/home/manu/repos/sourced/libuast/tests/../src/uast_private.h:20:18: warning: ‘xmlDoc* create_document(node_api*, void*)’ used but never defined
 static xmlDocPtr create_document(node_api *api, void *node);
```